## 1) Import the pandas library

In [27]:
import pandas as pd

## 2) Read in the file "netflix_titles.csv" and call the corresponding data frame netflix

---



In [28]:
path = "https://raw.githubusercontent.com/casbdai/notebooks2023/main/Module3/00_Onboarding/04_EntryExam/netflix_titles.csv"

netflix = pd.read_csv(path)
# what's the purpose of the below line, if the next code cell has .head() and .info()? Commented it out for now.
# netflix._____

## 3) Use the .head() and the .info() methods to explore the structure of the dataset

Answer the following questions:
- How many features has the netflix dataframe?
- How many instance has the dataframe?
- What are the datatypes of the dataframe? What do the different data types means?
- Which variables have missing values?


In [29]:
netflix.head()

show_id     type                                    title  \
0  81145628    Movie  Norm of the North: King Sized Adventure   
1  80117401    Movie               Jandino: Whatever it Takes   
2  70234439  TV Show                       Transformers Prime   
3  80058654  TV Show         Transformers: Robots in Disguise   
4  80125979    Movie                             #realityhigh   

                   director  \
0  Richard Finn, Tim Maltby   
1                       NaN   
2                       NaN   
3                       NaN   
4          Fernando Lebrija   

                                                cast  \
0  Alan Marriott, Andrew Toth, Brian Dobson, Cole...   
1                                   Jandino Asporaat   
2  Peter Cullen, Sumalee Montano, Frank Welker, J...   
3  Will Friedle, Darren Criss, Constance Zimmer, ...   
4  Nesta Cooper, Kate Walsh, John Michael Higgins...   

                                    country         date_added  release_year  \
0  United States, India, South Korea, China  September 9, 2019          2019   
1                            United Kingdom  September 9, 2016          2016   
2                             United States  September 8, 2018          2013   
3                             United States  September 8, 2018          2016   
4                             United States  September 8, 2017          2017   

     rating  duration                           listed_in  \
0     TV-PG    90 min  Children & Family Movies, Comedies   
1     TV-MA    94 min                     Stand-Up Comedy   
2  TV-Y7-FV  1 Season                            Kids' TV   
3     TV-Y7  1 Season                            Kids' TV   
4     TV-14    99 min                            Comedies   

                                         description  
0  Before planning an awesome wedding for his gra...  
1  Jandino Asporaat riffs on the challenges of ra...  
2  With the help of three human allies, the Autob...  
3  When a prison ship crash unleashes hundreds of...  
4  When nerdy high schooler Dani finally attracts...

In [30]:
netflix.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6234 entries, 0 to 6233
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       6234 non-null   int64 
 1   type          6234 non-null   object
 2   title         6234 non-null   object
 3   director      4265 non-null   object
 4   cast          5664 non-null   object
 5   country       5758 non-null   object
 6   date_added    6223 non-null   object
 7   release_year  6234 non-null   int64 
 8   rating        6224 non-null   object
 9   duration      6234 non-null   object
 10  listed_in     6234 non-null   object
 11  description   6234 non-null   object
dtypes: int64(2), object(10)
memory usage: 584.6+ KB


- 12 features
- 6234 instances
- the datatypes are: int64 and object
- the following variables have missing values: `director`, `cast`,`country`, `date_added` and `rating` 

## 4) What is the minimum, maximum and mean of the feature "release_year" ?

In [31]:
print("Min: " + str(netflix["release_year"].min()))

Min: 1925


In [32]:
print("Max: " + str(netflix["release_year"].max()))

Max: 2020


In [33]:
print("Mean: " + str(netflix["release_year"].mean()))

Mean: 2013.3593198588387


## 5) Select all netflix shows that have been released in 2020? How many are these?

In [34]:
netflix[netflix["release_year"]==2020].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 25 entries, 1315 to 3573
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       25 non-null     int64 
 1   type          25 non-null     object
 2   title         25 non-null     object
 3   director      8 non-null      object
 4   cast          23 non-null     object
 5   country       18 non-null     object
 6   date_added    25 non-null     object
 7   release_year  25 non-null     int64 
 8   rating        25 non-null     object
 9   duration      25 non-null     object
 10  listed_in     25 non-null     object
 11  description   25 non-null     object
dtypes: int64(2), object(10)
memory usage: 2.5+ KB


There are 25 shows released in 2020.

## 6) Create two new dataframes calles "netflix_years" and "netflix_titles". These datasets shall only contain the features "show_id" and "release_year" or "title"

In [35]:
netflix_titles = netflix[["show_id","title"]]
netflix_years = netflix["release_year"]

## 7) Merge the two dataframes together using pd.concat() and pd.merge(). What is the difference between these two approaches?

In [36]:
netflix_concat = pd.concat([netflix_titles, netflix_years], axis=1)
netflix_concat

show_id                                        title  release_year
0     81145628      Norm of the North: King Sized Adventure          2019
1     80117401                   Jandino: Whatever it Takes          2016
2     70234439                           Transformers Prime          2013
3     80058654             Transformers: Robots in Disguise          2016
4     80125979                                 #realityhigh          2017
...        ...                                          ...           ...
6229  80000063                                 Red vs. Blue          2015
6230  70286564                                        Maron          2016
6231  80116008       Little Baby Bum: Nursery Rhyme Friends          2016
6232  70281022  A Young Doctor's Notebook and Other Stories          2013
6233  70153404                                      Friends          2003

[6234 rows x 3 columns]

In [37]:
netflix_merge = pd.merge(netflix_titles, netflix_years, on=netflix_titles["show_id"])
netflix_merge

key_0   show_id                                        title  \
0     81145628  81145628      Norm of the North: King Sized Adventure   
1     80117401  80117401                   Jandino: Whatever it Takes   
2     70234439  70234439                           Transformers Prime   
3     80058654  80058654             Transformers: Robots in Disguise   
4     80125979  80125979                                 #realityhigh   
...        ...       ...                                          ...   
6229  80000063  80000063                                 Red vs. Blue   
6230  70286564  70286564                                        Maron   
6231  80116008  80116008       Little Baby Bum: Nursery Rhyme Friends   
6232  70281022  70281022  A Young Doctor's Notebook and Other Stories   
6233  70153404  70153404                                      Friends   

      release_year  
0             2019  
1             2016  
2             2013  
3             2016  
4             2017  
...            ...  
6229          2015  
6230          2016  
6231          2016  
6232          2013  
6233          2003  

[6234 rows x 4 columns]

pd.concat() copies the datasets together. If axis = 0 the datasets are copied row-wise together. If axis = 1 it is column-wise.

pd.merge() merges together dataframes on a key (=show_id). Thus, it mantains the connection between the same instances (even after sorting etc.). pd.concat does not mantain the logic between instances. If one data frame is sorted ascending and the other one descending, the connection between the instances is destroyed.

## 8) Write the "netflix_merge" dataframe to your google drive / local harddisk as csv file

### Google Collab (Save file to Google Drive)

*Note: I'm not using Colab so I didn't complete this part* 

**Connect to google collab with jupyter notebook to google drive**

In [38]:
# from google.colab import drive
# drive.mount('/content/drive')

**We need to adapt the path to which we want to save the file to your folder structure in your google drive**

remember:
- *drive/MyDrive* is the standard prefix to start-
- *YOURFOLDER1/YOURFOLDER2/* resembles your folder structure that you created in google drive (you need to create the folders first)
- the path always has to end with */* because we are adding the file name in a second step

In [39]:
# path = "drive/My Drive/YOURFOLDER1/YOURFOLDER2/" #please adapt

# # This is how the path looks at my google drive:
# # path = "drive/MyDrive/Colab Notebooks/CAS BDAI/PrepTasks Module 3/EntryExam"

now we can save the file with pandas using the .to_csv() method

In [40]:
# netflix_merge.to_csv(path + "netflix_merge.csv")

### Anaconda (Save file to your local hard disk)

*Note: I'm using VS Code, but shouldn't make any difference*



**We need to adapt the path to which we want to save the file to your folder structure on your local hard disk**

remember:
- please change the path to the folder in which you want to save the file below such that it matches the folder structure that you created on your hard disk (you need to create the folders first)
- the path always has to end with */* because we are adding the file name in a second step

In [41]:
path = "/home/jean/Documents/dev/cas-learn/learning-module3/data/" #please adapt


now we can save the file with pandas using the .to_csv() method

In [42]:
netflix_merge.to_csv(path + "netflix_merge.csv")